clang/lib/Sema/SemaExpr.cpp

```diff
@@ -20224,6 +20224,8 @@ void Sema::DiagnoseEqualityWithExtraParens(ParenExpr *ParenE) {
     return;
 
   Expr *E = ParenE->IgnoreParens();
+  if (ParenE->isProducedByFoldExpansion() && ParenE->getSubExpr() == E)
+    return;
```

Comment:
So we would expect htis to still diagnose in a case where:

`if (((a ==b )) or ...`

.  Note hte extra parens there.  Does this still work?  My point being, perhaps we have to strip off this set of parens before calling this function instead.

Reply:
Woops, i see that it is in the tests, so, LGTM!

Reply:
@erichkeane thanks for the review.